projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml

```
@@ -346,6 +374,56 @@ data:
                   - name: envoy.filters.http.router
                     typed_config:
                       "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
+{{- if $statsConfig.enabled }}
+      - name: prometheus_listener
+        address:
+          socket_address:
+            address: 0.0.0.0
+            port_value: 9091
+        filter_chains:
+          - filters:
+              - name: envoy.filters.network.http_connection_manager
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+                  codec_type: AUTO
+                  stat_prefix: prometheus
+                  route_config:
+                    name: prometheus_route
+                    virtual_hosts:
+                      - name: prometheus_host
+                        domains:
+                          - "*"
+                        routes:
+                          - match:
+                              path: "/ready"
+                              headers:
+                                - name: ":method"
+                                  exact_match: GET
+                            route:
+                              cluster: admin_port_cluster
+                          - match:
+                              prefix: "/metrics"
+                              headers:
+                                - name: ":method"
+                                  exact_match: GET
+                            route:
+                              prefix_rewrite: {{ $statsConfig.routePrefixRewrite }}
+                              cluster: admin_port_cluster
+                        {{- if $statsConfig.enableStatsRoute}}
+                          - match:
+                              prefix: "/stats"
+                              headers:
+                                - name: ":method"
+                                  exact_match: GET
+                            route:
+                              prefix_rewrite: {{ $statsConfig.statsPrefixRewrite }}
```

I think this should be 
```suggestion
                              prefix_rewrite: {{ $statsConfig.statsRoutePrefixRewrite }}
```

---

Oh nope looks like there is a subtle difference in the naming but it is correct here

---

FWIW this is a true cherry pick, i.e. this is just the commit taken from main